src/Provider/StandardUser.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace League\OAuth2\Client\Provider;
+
+class StandardUser implements UserInterface
+{
+    /**
+     * @var array
+     */
+    protected $response;
+
+    /**
+     * @var string
+     */
+    protected $uid;
+
+    public function __construct(array $response, $uid)
+    {
+        $this->response = $response;
+        $this->uid = $uid;
+    }
+
+    public function getUserId()
+    {
+        return $this->response[$this->uid];
+    }
+
+    /**
+     * Get the raw user response.
+     *
+     * @return array
+     */
+    public function toArray()
```

Comment:
We really don't know what additional user details there will be for a generic provider. User details aren't part of OAuth 2.0, so this is perfect, I think.